presto-spi/src/main/java/io/prestosql/spi/connector/AggregationApplicationResult.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import io.prestosql.spi.expression.ConnectorExpression;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class AggregationApplicationResult<T>
+{
+    private final T handle;
+    private final List<ConnectorExpression> projections;
+    private final List<Assignment> assignments;
+    private final Map<ColumnHandle, ColumnHandle> groupingSetMapping;
+
+    public AggregationApplicationResult(
+            T handle,
+            List<ConnectorExpression> projections,
+            List<Assignment> assignments,
+            Map<ColumnHandle, ColumnHandle> groupingSetMapping)
+    {
+        this.handle = requireNonNull(handle, "handle is null");
+        requireNonNull(groupingSetMapping, "goupingSetMapping is null");
+        requireNonNull(projections, "projections is null");
+        requireNonNull(assignments, "assignment is null");
+        this.groupingSetMapping = Map.copyOf(groupingSetMapping);
+        this.projections = List.copyOf(projections);
+        this.assignments = List.copyOf(assignments);
+    }
+
+    public T getHandle()
+    {
+        return handle;
+    }
+
+    public List<ConnectorExpression> getProjections()
+    {
+        return projections;
+    }
+
+    public List<Assignment> getAssignments()
+    {
+        return assignments;
+    }
+
+    public Map<ColumnHandle, ColumnHandle> getGroupingSetMapping()
```

Comment:
Let's call this `groupingColumnMapping`. Grouping sets are made of sets of columns, but this method is not specifically about the grouping sets themselves.

Reply:
Done.

Reply:
@Parth-Brahmbhatt @martint what is groupingColumnMapping used for ? Thanks.